lib/Net/SSLeay.pod

```diff
@@ -4625,6 +4625,32 @@ When callback is undef, an existing callback will be disabled.
 
 Check openssl doc L<http://www.openssl.org/docs/ssl/SSL_CTX_set_info_callback.html|http://www.openssl.org/docs/ssl/SSL_CTX_set_info_callback.html>
 
+=item * set_msg_callback
+
+Sets the callback function, that can be used to obtain protocol messages information for $ssl during connection setup and use.
+When callback is undef, the callback setting currently valid for ctx is used.
+
+ Net::SSLeay::set_msg_callback($ssl, $cb, [$data]);
```

Comment:
OpenSSL uses a separate function to set the `arg` parameter for the callback function:
 `void SSL_CTX_set_msg_callback_arg(SSL_CTX *ctx, void *arg);`

Is there a reason to not use or expose `SSL_CTX_set_msg_callback_arg`?

Reply:
Doing some testing and now it feels like using `set_msg_callback()` to set `$data` seems to be easier for handling reference counting. When the function is called with `undef` callback, it will also take care of `$data`.

If this is the case, can you update the documentation to say that `SSL_CTX_set_msg_callback_arg()` is not needed for this reason.

Reply:
ok, I'll update the documentation. I just didn't want to confuse people by just not referencing SSL_CTX_set_msg_callback_arg as it's not needed.

However, how that $data is used in SSLeay.xs with a global variable to store this data ref and callback subroutine is a bit strange to me. The data ptr can be used to store a possible callback data SV value in theory, but that would change the way this callback works and this code is overall just a copy-paste of the other callback methods that are already there. Else, if this would be mine, I think I would change this - but then I'd change all callbacks and that's totally out of scope of course.

Reply:
I'd say it's fine to mention the unneeded function because we link to OpenSSL's documentation as our primary source. When someone checks the link, they see that there's a function that we don't have available and also say nothing about. But if we just say that use `$data` when setting the callback, and `SSL_*set_msg_callback_arg` is not needed, that hopefully clarifies it.

Regarding use of callback argument, I tested the message callback with a logging related object as callback `$data`. The current behaviour seems to work nice because it takes care of reference counting and the object did not get garbage collected until I called `CTX_set_msg_callback` with `$cb` set to `undef`.